pkg/istio-agent/sds-agent.go

```diff
@@ -208,11 +220,16 @@ type SDSAgent struct {
 //
 // If node agent and JWT are mounted: it indicates user injected a config using hostPath, and will be used.
 //
-func NewSDSAgent(discAddr string, tlsRequired bool, pilotCertProvider, jwtPath, outputKeyCertToDir, clusterID string) *SDSAgent {
+func NewSDSAgent(discAddr string, tlsRequired bool, pilotCertProvider, jwtPath, outputKeyCertToDir string,
```

Comment:
nit: if we are passing in all of these ares already, maybe we should pass in a partial SDSAgent and then  this method can be called FillInDefaults or something and add in all the logic? Otherwise the function signature just becomes huge

Reply:
+1 - use a struct.

Reply:
Or better: pass ProxyConfig, where those things should be added.